src/hdmf/backends/hdf5/h5_utils.py

```diff
@@ -140,7 +140,10 @@ def stringify(spec):
 
     def __write(self, d, name):
         data = self.stringify(d)
-        dset = self.__group.create_dataset(name, data=data, dtype=self.__str_type)
+        if name in self.__group:  # handle append mode
```

Comment:
This behavior is handled by [require_dataset](http://docs.h5py.org/en/stable/high/group.html?highlight=require_dataset#Group.require_dataset)